test/acceptance/tests/mesh-gateway/mesh_gateway_test.go

```diff
@@ -225,23 +225,17 @@ func TestMeshGatewaySecure(t *testing.T) {
 // verifyFederation checks that the WAN federation between servers is successful
 // by first checking members are alive from the perspective of both servers.
 // If secure is true, it will also check that the ACL replication is running on the secondary server.
-func verifyFederation(t *testing.T, primaryClient, secondaryClient *api.Client, secure bool) {
-	const consulMemberStatusAlive = 1
-
-	retrier := &retry.Timer{Timeout: 1 * time.Minute, Wait: 1 * time.Second}
+func verifyFederation(t *testing.T, primaryClient, secondaryClient *api.Client, releaseName string, secure bool) {
+	retrier := &retry.Timer{Timeout: 2 * time.Minute, Wait: 1 * time.Second}
 
 	retry.RunWith(retrier, t, func(r *retry.R) {
-		members, err := primaryClient.Agent().Members(true)
+		secondaryServerHealth, _, err := primaryClient.Health().Node(fmt.Sprintf("%s-consul-server-0", releaseName), &api.QueryOptions{Datacenter: "dc2"})
```

Comment:
Was just curious what this line does, is it the consul client on the primary datacenter is checking the health endpoint of the server in secondary datacenter? Why do we need to make this request from opposite datacenters?

Reply:
Sorry! I will add a comment.

Yup exactly. We're trying to see if one datacenter can "see" the other, and vice versa. For that, we calling the health API to check the health of the server in dc1 from the perspective of dc2 and vice versa. This is how we now the federation is "successful" because that means one datacenter can query and forward requests to the other.

Reply:
Ahh, got it. Thank you!

Reply:
Added a comment! Thanks for the question 🙏